src/Turbo/src/Doctrine/ClassUtil.php

```diff
@@ -22,12 +22,7 @@ final class ClassUtil
     public static function getEntityClass(object $entity): string
     {
         if ($entity instanceof LazyObjectInterface) {
-            $entityClass = get_parent_class($entity);
-            if (false === $entityClass) {
-                throw new \LogicException('Parent class missing');
-            }
-
-            return $entityClass;
+            return get_parent_class($entity) ?: $entity::class;
```

Comment:
If something implements `LazyObjectInterface` but it doesn't have a parent class, does returning `$entity::class` make sense? Or is that a misconfiguration of some sort (when *would* this really occur?)?

Reply:
it does yes, the interface doesn't have to be implemented by a proxy
this is mostly theory in the context of doctrine of course :)